addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -0,0 +1,426 @@
+            'extraDiscountAmount': 0.0,
+            'totalItemsDiscountAmount': 0.0,
+        })
+        return eta_invoice
+
+    def _l10n_eg_eta_prepare_invoice_lines_data(self, invoice):
+        lines = []
+        for line in invoice.invoice_line_ids.filtered(lambda x: x.display_type not in ('line_section', 'line_note')):
+            discount = (line.discount / 100.0) * line.quantity * line.price_unit
+            lines.append({
+                'description': line.name,
+                'itemType': line.product_id.barcode.startswith('EG') and 'EGS' or 'GS1',
+                'itemCode': line.product_id.barcode,
+                'unitType': line.product_uom_id.l10n_eg_unit_code_id.code,
+                'quantity': line.quantity,
+                'internalCode': line.product_id.default_code or '',
+                'salesTotal': invoice._get_amount_main_currency(line.quantity*line.price_unit),
+                'total': invoice._get_amount_main_currency(line.price_total),
+                'valueDifference': 0.0,
+                'totalTaxableFees': 0.0,
+                'netTotal': invoice._get_amount_main_currency(line.price_subtotal),
+                'itemsDiscount': 0.0,
+                'unitValue': {
+                    'currencySold': invoice.currency_id.name,
+                    'amountEGP': invoice._get_amount_main_currency(line.price_unit)
+                },
+                'discount': {  # price_discount
+                    'rate': line.discount or 0.0,
+                    'amount': invoice._get_amount_main_currency(discount) or 0.0,
+                },
+                'taxableItems': [
+                    {
+                        'taxType': tax.l10n_eg_eta_code.split('_')[0].upper(),
+                        'amount': abs(invoice._get_amount_main_currency((tax.amount / 100.0) * line.price_subtotal)) or 0,
+                        'subType': tax.l10n_eg_eta_code.split('_')[1].upper(),
+                        'rate': abs(tax.amount)
+                        or 0.0,
+                    }
+                    for tax in line.tax_ids
+                ],
+            }
+            )
+            # if the base currency isn't EGP
+            if invoice.currency_id != self.env.ref('base.EGP'):
+                lines[-1]['unitValue']['currencyExchangeRate'] = invoice._exchange_currency_rate()
+                lines[-1]['unitValue']['amountSold'] = line.price_unit
+        return lines
+
+    def _l10n_eg_get_partner_tax_type(self, partner_id, issuer=False):
+        if issuer:
+            return 'B'
+        elif partner_id.commercial_partner_id.country_code == 'EG':
+            return 'B' if partner_id.commercial_partner_id.is_company else 'P'
+        else:
+            return 'F'
+
+    def _l10n_eg_eta_prepare_address_data(self, partner_id, issuer=False, invoice_amount=False):
+        address = {
+            'address': {
+                'country': partner_id.country_id.code,
```

Comment:
partner_id would suggest it is an id.  Just partner might be clearer.  
```suggestion
                'country': partner.country_id.code,
```